src/test/java/com/iota/iri/service/tipselection/impl/ConnectedComponentsCalculatorTest.java

```diff
@@ -0,0 +1,240 @@
+
+        Assert.assertTrue(recentTransactions.contains(transaction.getHash()));
+        Assert.assertTrue(recentTransactions.contains(Hash.NULL_HASH));
+
+    }
+
+    @Test
+    public void returnsGenesisTxInChain() throws Exception {
+
+        long incr = 1000;
+        final int chainLength = ConnectedComponentsCalculator.N - 10;
+        List<TransactionViewModel> transactions = new ArrayList<TransactionViewModel>();
+
+        transactions.add(new TransactionViewModel(getRandomTransactionTrits(), getRandomTransactionHash()));
+
+        for (int i = 0; i < chainLength; i++) {
+            Hash prevTxHash = transactions.get(transactions.size() - 1).getHash();
+            transactions.add(new TransactionViewModel(
+                    getRandomTransactionWithTrunkAndBranch(prevTxHash, prevTxHash), getRandomTransactionHash()));
+        }
+
+        for (TransactionViewModel transaction : transactions) {
+            transaction.setArrivalTime(incr++);
+            transaction.store(tangle);
+        }
+
+        Mockito.when(tipsViewModel.getRandomSolidTipHash()).thenReturn(transactions.get(transactions.size() - 1).getHash());
+
+        ConnectedComponentsCalculator connectedComponentsCalculator = new ConnectedComponentsCalculator(tangle);
+        Collection<Hash> recentTransactions = connectedComponentsCalculator.findNMostRecentTransactions(
+                Collections.singleton(tipsViewModel.getRandomSolidTipHash()));
+        Assert.assertTrue(recentTransactions.contains(Hash.NULL_HASH));
+    }
+
+    @Test
+    public void doesntReturnsGenesisTxInChain() throws Exception {
+        long incr = 1000;
+
+        final int chainLength = ConnectedComponentsCalculator.N + 1;
+        List<TransactionViewModel> transactions = new ArrayList<TransactionViewModel>();
+
+        transactions.add(new TransactionViewModel(getRandomTransactionTrits(), getRandomTransactionHash()));
+
+        for (int i = 0; i < chainLength; i++) {
+            Hash prevTxHash = transactions.get(transactions.size() - 1).getHash();
+            transactions.add(new TransactionViewModel(
+                    getRandomTransactionWithTrunkAndBranch(prevTxHash, prevTxHash), getRandomTransactionHash()));
+        }
+
+        for (TransactionViewModel transaction : transactions) {
+            transaction.setArrivalTime(incr++);
+            transaction.store(tangle);
+        }
+
+        Mockito.when(tipsViewModel.getRandomSolidTipHash()).thenReturn(transactions.get(transactions.size() - 1).getHash());
+
+        ConnectedComponentsCalculator connectedComponentsCalculator = new ConnectedComponentsCalculator(tangle);
+        Collection<Hash> recentTransactions = connectedComponentsCalculator.findNMostRecentTransactions(
+                Collections.singleton(tipsViewModel.getRandomSolidTipHash()));
+        Assert.assertFalse(recentTransactions.contains(Hash.NULL_HASH));
+    }
+
+    @Test
+    public void starAroundGenesis() throws Exception {
```

Comment:
change name to reflect what's being tested (`allTipsAndGenesisReturnedForStarAroundGenesis`)